app/models/concerns/twitch_user.rb

```diff
@@ -5,9 +5,20 @@ module TwitchUser
   extend ActiveSupport::Concern
 
   included do
-    def follows
-      Rails.cache.fetch([:db, :follows, self], expires_in: 1.hour) do
-        User.where(twitch_id: Twitch::Follows.followed_ids(twitch_id)).joins(:runs).group('users.id')
+    def sync_twitch_follows!
+      ActiveRecord::Base.transaction do
+        current_followed_users = User.where(twitch_id: Twitch::Follows.followed_ids(twitch_id))
+        old_followed_users = twitch_followed_users
+
+        (current_followed_users - old_followed_users).each do |u|
+          TwitchUserFollow.create!(from_user: self, to_user: u)
+        end
+
+        (old_followed_users - current_followed_users).each do |u|
+          TwitchUserFollow.find(from_user: self, to_user: u).destroy
+        end
+
+        update(twitch_user_follows_checked_at: Time.now.utc)
```

Comment:
If we kick off this job and the user visits the frontpage again before we finish, we'll end up kicking off another identical job. Haven't thought through possible race conditions if that happens but they seem possible. We might want to make this timestamp update before anything, to prevent oddness there.